src/Api/MediaLibraryApi.php

```diff
@@ -160,14 +160,96 @@ public function mediaPackagePackageNameGet(string $package_name, ?int $limit = 2
    */
   public function mediaFileIdGet(int $id, &$responseCode, array &$responseHeaders)
   {
-    $responseCode = Response::HTTP_I_AM_A_TEAPOT;
+    $json_response_array = [];
+    $media_package_file = $this->entity_manager->getRepository(MediaPackageFile::class)
+      ->findOneBy(['id' => $id])
+        ;
+
+    if (null === $media_package_file)
+    {
+      $responseCode = Response::HTTP_NOT_FOUND; // => Not found
+
+      return null;
+    }
+
+    $responseCode = Response::HTTP_OK; // 200 => OK
+
+    /** @var MediaPackageFile $media_package_file */
+    $mediaFile = [
+      'id' => $media_package_file->getId(),
+      'name' => $media_package_file->getName(),
+      'flavor' => $media_package_file->getFlavor(),
+      'package' => $media_package_file->getCategory()->getPackage()->first()->getName(),
+      'category' => $media_package_file->getCategory()->getName(),
+      'author' => $media_package_file->getAuthor(),
+      'extension' => $media_package_file->getExtension(),
+      'download_url' => $this->url_generator->generate(
+          'download_media',
+          ['id' => $media_package_file->getId()],
+          UrlGenerator::ABSOLUTE_URL),
+    ];
+    $json_response_array[] = new MediaFile($mediaFile);
+
+    $responseData = new MediaFiles(['media_files' => $json_response_array, 'total_results' => 1]);
+
+    return $responseData;
   }
 
   /**
    * {@inheritdoc}
    */
   public function mediaFilesGet(?int $limit = 20, ?int $offset = 0, ?string $flavor = null, &$responseCode, array &$responseHeaders)
   {
-    $responseCode = Response::HTTP_I_AM_A_TEAPOT;
+    if (null === $limit)
+    {
+      $limit = 20;
+    }
+    if (null === $offset)
+    {
+      $offset = 0;
+    }
+
+    $total_results = 0;
+
+    $json_response_array = [];
+
+    $media_package_files = $this->entity_manager->getRepository(MediaPackageFile::class)
+      ->findAll()
+        ;
+
+    if (null !== $media_package_files)
+    {
+      /** @var MediaPackageFile $media_package_file */
+      foreach ($media_package_files as $media_package_file)
+      {
+        if ((0 != $offset && count($media_package_files) <= $offset) || count($json_response_array) === $limit)
```

Comment:
might reduce the compelxity/nesting here + any reason why no strict comparison was applied between 0 != offset?

```
if ((0 !== $offset && count($media_package_files) <= $offset))
{
  $offset -= count($media_package_files);
  continue;
}

if (count($json_response_array) === $limit)
{
  continue;
}
```